gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/distributed/message/StatementExecutorMessage.java

```diff
@@ -146,7 +146,10 @@ public StatementExecutorMessage(ResultCollector<Object, T> collector,
       long timeOutMillis, boolean abortOnLowMemory) {
     super(collector, region, routingObjects, getCurrentTXState(lcc),
         getTimeStatsSettings(lcc), abortOnLowMemory);
-    if (source != null && !Property.DEFAULT_USER_NAME.equals(defaultSchema)) {
+    // skip setting defaultSchema when it is for APP system user
+    if (source != null && !(Property.DEFAULT_USER_NAME.equals(defaultSchema) &&
```

Comment:
Do we see a situation where in a secured snappy cluster can have even default schema being used? I mean an authenticated user's default schema will be his username space only. Skipping it is fine but I don't think this will actually be the case. Right? If yes then probably we can just set the default schema blindly.

Reply:
This is for an unsecured cluster where the user is APP and then the defaultSchema can be skipped because that is what remote connection will used.

Earlier remote side (GfxdConnectionWrapper) would always use peer authentication service properties for the connection but that was a problem and now it uses the dbOwner instead (which can be different if peer authentication service is different from normal one).